src/nile/utils/debug.py

```diff
@@ -0,0 +1,89 @@
+"""Functions used to help debug a rejected transaction."""
+
+import json
+import logging
+import os
+import re
+import subprocess
+import time
+
+from nile.common import BUILD_DIRECTORY, DEPLOYMENTS_FILENAME, GATEWAYS
+
+
+def locate_error(tx_hash, network, contracts_file=None):
+    """Use available contracts to help locate the error in a rejected transaction."""
+    command = ["starknet", "tx_status", "--hash", tx_hash]
+
+    if network == "mainnet":
+        os.environ["STARKNET_NETWORK"] = "alpha-mainnet"
+    elif network == "goerli":
+        os.environ["STARKNET_NETWORK"] = "alpha-goerli"
+    else:
+        command.append(f"--gateway_url={GATEWAYS.get(network)}")
+
+    logging.info(
+        "⏳ Querying the network to check transaction status and identify contracts..."
+    )
+
+    while True:
+        receipt = json.loads(subprocess.check_output(command))
+        status = receipt["tx_status"]
+        if status == "REJECTED":
+            break
+        output = f"Transaction status: {status}"
+        if status.startswith("ACCEPTED"):
+            logging.info(f"✅ {output}. No error in transaction.")
+            return
+        logging.info(f"🕒 {output}. Trying again in 30 seconds unless stopped.")
+        time.sleep(30)
```

Comment:
[nice to have] we might want to abstract this away to a variable (maybe `common.py`?) so it's easy to make it configurable in the future